### PR TITLE
#17 Add multi-language support 

### LINKED ISSUE #17: Add multi-language support
Use `flutter_localizations` / `l10n` for adding multi-language support. Currently hardcoded strings are scattered throughout the app, need to find them and translations.

By default it should use language as device's language. If the device's language is not supported, English should be used.

### PR DESCRIPTION
**Pull request by bibekpanda55**
## What type of PR is this? (check all applicable)

- [ ] Feature

## Featured Covered in this PR

- [ ] Added multi-language support #17

## Related Tickets & Documents

- Related Issue #17 
- Closes #17

### CODE REVIEW

File: lib/core/widgets/home_page_app_bar.dart [hunk 1]
```diff
@@ -152,7 +153,7 @@ class Action extends StatelessWidget {
       Widget getSuitableWidget() {
         if (isSearchEnabled) {
           return Padding(
-            key: const ValueKey("close icon"),
```
Comment (SankethBK) [inline lib/core/widgets/home_page_app_bar.dart:155]: No need to change this text as its for `ValueKey`. Its only for identifying widgets, this text won't be visible to users

File: lib/core/widgets/home_page_app_bar.dart [hunk 2]
```diff
@@ -174,7 +175,7 @@ class Action extends StatelessWidget {
           );
         } else {
           return Padding(
-            key: const ValueKey("search icon"),
+            key: ValueKey(AppLocalizations.of(context).searchIcon),
```
Comment (SankethBK) [inline lib/core/widgets/home_page_app_bar.dart:178]: Same as [this](https://github.com/SankethBK/diaryvault/pull/74/files#r1349925553)

Comment (bibekpanda55) [reply lib/core/widgets/home_page_app_bar.dart:178]: I didn't change it at first commit. Then in the second commit I updated it. No issues will revert it back and make a new PR

Comment (SankethBK) [reply lib/core/widgets/home_page_app_bar.dart:178]: don't create new PR it would be difficult to manage, add a commit to this PR only

Comment (bibekpanda55) [reply lib/core/widgets/home_page_app_bar.dart:178]: Got it will do.

File: lib/l10n/arb/app_en.arb [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+"lastSynced":"Last synced: ",
+"notAvailable":"Not available",
+"googleDrive":"Google Drive",
+"chooseTheSyncSource":"Choose the Sync Source",
+"done":"Done",
+"delete":"Delete",
+"cancel":"Cancel",
+"submit":"Submit",
+"appDescription":"Discover diaryVault - a diary app designed to help you capture your thoughts, memories, and moments effortlessly. Available now on the Play Store!"
+
```
Comment (SankethBK) [inline lib/l10n/arb/app_en.arb:81]: Format this file like a json file
```
{
  "language":"English",
  "@language":{
    "description":"The current Language"
  },
  
  "closeIcon":"close icon",
  "searchIcon":"search icon",
  "pageNotFound":"Page not found",
  "settings":"Settings",
  "enterNewEmail":"Enter new email"
  ...
```

Comment (bibekpanda55) [reply lib/l10n/arb/app_en.arb:81]: Didn't get you? It is formatted as a json file already

Comment (SankethBK) [reply lib/l10n/arb/app_en.arb:81]: Add one level of indentation for its contents
image

File: lib/l10n/arb/app_en.arb [hunk 4]
```diff
@@ -0,0 +1,82 @@
+{
+"language":"English",
+"@language":{
+  "description":"The current Language"
+},
+
+"closeIcon":"close icon",
```
Comment (SankethBK) [inline lib/l10n/arb/app_en.arb:7]: you can remove this as we removed `closeIcon` from `ValueKey`

Comment (bibekpanda55) [reply lib/l10n/arb/app_en.arb:7]: Yes got it.

File: lib/l10n/arb/app_en.arb [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+"searchIcon":"search icon",
```
Comment (SankethBK) [inline lib/l10n/arb/app_en.arb:8]: remove this as well as we removed`searchIcon` from `ValueKey`

Comment (bibekpanda55) [reply lib/l10n/arb/app_en.arb]: Have added indentation and removed unused keys and made a commit for this file. You can check

**Review by SankethBK — APPROVED** [review]
LGTM, thanks for contributing

File: lib/l10n/arb/app_en.arb (hunk 5 above)
Comment (bibekpanda55) [reply lib/l10n/arb/app_en.arb]: You can comment if you need any more modifications.

**Pull request merged**